.github/workflows/build.yml

```
@@ -0,0 +1,38 @@
+name: build
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        build_type: [Debug, Release]
+    
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Ubuntu Packages
+      if: matrix.os == 'ubuntu-latest'
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libassimp-dev libglfw3-dev
+
+    - name: Install MacOS Packages
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      run: |
+        brew install assimp
+        brew install glfw
+
+    - name: Generate
+      shell: bash
+      run: cmake -Bbuild -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
```

```suggestion
      run: |
        [[ ${{ matrix.os }} = 'macos-latest' ]] && generator='Xcode' || generator='Ninja'
        cmake -G"${generator}" -B${{ github.workspace }}/build -DCMAKE_BUILD_TYPE=${{ matrix.build_type }}
```
1. If a generator is not specified, "Unix Makefiles" will be used on macOS whereas most students will be using Xcode. On Linux, ninja should provide better build performance as it will be using multiple threads by default unlike make. As for Visual Studio, if using the built-in CMake support (which we recommend students to use), it actually uses ninja underneath rather than MSbuild so we should rather test that path.
2. This is the path used in the examples provided by GitHub, which might be safer in case they move the _build/_ folder around.